src/main/java/Sb2.java

```diff
@@ -23,24 +25,57 @@ public Sb2(String filePath) throws IOException {
 
         String pathToPackageJson = destPath + File.separator + "package.json";
         String jsonString = getFileContents(pathToPackageJson);
-        sb2Json = createJSONObject(jsonString);
+        init(createJSONObject(jsonString));
+    }
+    /**
+     * Construct an Sb2 object from a JSONObject. Useful for testing.
+     * @param stage JSONObject from which to construct Sb2
+     */
+    public Sb2(JSONObject stage) {
+        init(stage);
     }
     /**
+     * Function to be called from all constructors.
+     * @param stage JSONObject which is underlying data structure for Sb2.
+     */
+    public void init(JSONObject stage) {
+        this.stage = stage;
+        this.sprites = new Sprites(stage);
+    }
+
+    /**
      * Return underlying JSONObject.
      * @return The underlying JSONObject.
      */
     public JSONObject getJSONObject() {
-        return sb2Json;
+        return stage;
     }
 
     /**
-     * Return the number of Sprites in Sb2 object.
-     * @return The number of sprites in project
+     * Each sprite has a unique name.
+     * This method returns these names in a array of strings.
+     * @return Array of sprite names.
      */
-    public int countSprites() {
-        return 0;
+    public String[] getSpriteNames() {
```

Comment:
You're right, of course. This is more maintainable. I've seen composition handled somewhat differently in the past and seeing it like this confused me somewhat because it introduced extra methods. That being said, it has a clear benefit. Thanks for the link!